### PR TITLE
[WIP] Lims and Login

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Dear all,

Started with LIMS and Login, data structures for LIMS still to be defined.

Regards,
Marcus

### CODE REVIEW

File: lims.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    def get_proposal_by_username(str:username) -> List[LIMSProposal]:
+        """
+        :param str username: username
+        :returns: The list proposals belonging to username
+        :rtype: List[LIMSProposal]
+        """
+        pass
+
+    def get_proposal(str:code, str:number) -> LIMSProposal:
```
Comment (IvarsKarpics) [inline lims.rst:49]: Maybe get_proposal_by_code_and_number?

File: lims.rst [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    def get_sample(str: sample_id) -> LimsSample:
+        """
+        :param str sample_id: 
+        :returns: Sample with id <sample_id>
+        :rtype: LimsSample
+        """
+        pass
+
+    def get_task_url(str: task_id) -> str:
```
Comment (IvarsKarpics) [inline lims.rst:89]: It will return a link to ispyb, exi or synchweb? I guess the root of the path will come from the xml.

**Review by IvarsKarpics — APPROVED** [review]

File: lims.rst (hunk 1 above)
Comment (marcus-oscarsson) [reply lims.rst:49]: well, I thought that proposals are usually or traditionally identified by code and number and that get_proposal_by_code_and_number was slightly long. I have, however nothing against it per say.

File: lims.rst (hunk 2 above)
Comment (marcus-oscarsson) [reply lims.rst:89]: Yes, well it will return the link that the LIMS used returns ...

Comment (marcus-oscarsson) [reply lims.rst:89]: The idea is to return a url to the task within the LIMS application used, i.e. EXI.

**Pull request merged**